File: app/src/main/java/com/orange/tpms/ue/frag/Frag_setting_unit.java
package com.orange.tpms.ue.frag;

import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.de.rocket.ue.frag.RoFragment;
import com.orange.tpms.R;
import com.orange.tpms.adapter.SelectAdapter;
import com.orange.tpms.helper.UnitHelper;
import com.orange.tpms.widget.TitleWidget;

import com.de.rocket.ue.injector.BindView;

/**
 * 设置详情页
 * Created by haide.yin() on 2019/4/4 9:28.
 */
public class Frag_setting_unit extends Frag_base {

    @BindView(R.id.v_title_bar)
    TitleWidget twTitle;//Title
    @BindView(R.id.rv_temp)
    RecyclerView rvTemperature;//列表
    @BindView(R.id.rv_tire)
    RecyclerView rvTirePressure;//列表
    @BindView(R.id.rv_numeral)
    RecyclerView rvNumeral;//列表

    private SelectAdapter tempAdapter;//Adapter
    private SelectAdapter tireAdapter;//Adapter
    private SelectAdapter numeralAdapter;//Adapter
    private LinearLayoutManager tempLinearLayoutManager;//列表表格布局
    private LinearLayoutManager tireLinearLayoutManager;//列表表格布局
    private LinearLayoutManager numeralLinearLayoutManager;//列表表格布局
    private UnitHelper unitHelper;//Helper

    @Override
    public int onInflateLayout() {
        return R.layout.frag_setting_unit;
    }

    @Override
    public void initViewFinish(View inflateView) {
        initView();
        initHelper();
    }

    @Override
    public void onNexts(Object o) {

    }
    /**
     * 初始化页面
     */
    private void initView() {
        //设置标题
        twTitle.setTvTitle(R.string.app_unit);
        //返回
        twTitle.setOnBackListener((view) -> back());
        //配置TempRecyclerView
        if (tempLinearLayoutManager == null) {
            tempLinearLayoutManager = new LinearLayoutManager(activity);
        }
        rvTemperature.setLayoutManager(tempLinearLayoutManager);
        tempAdapter = new SelectAdapter(activity);
        rvTemperature.setAdapter(tempAdapter);
        tempAdapter.setOnItemClickListener((pos, content) ->{
            unitHelper.setTemp(activity,pos);
        });
        //配置TireRecyclerView
        if (tireLinearLayoutManager == null) {
            tireLinearLayoutManager = new LinearLayoutManager(activity);
        }
        rvTirePressure.setLayoutManager(tireLinearLayoutManager);
        tireAdapter = new SelectAdapter(activity);
        rvTirePressure.setAdapter(tireAdapter);
        tireAdapter.setOnItemClickListener((pos, content) ->{
            unitHelper.setPressure(activity,pos);
        });
        //配置NumeralRecyclerView
        if (numeralLinearLayoutManager == null) {
            numeralLinearLayoutManager = new LinearLayoutManager(activity);
        }
        rvNumeral.setLayoutManager(numeralLinearLayoutManager);
        numeralAdapter = new SelectAdapter(activity);
        rvNumeral.setAdapter(numeralAdapter);
        numeralAdapter.setOnItemClickListener((pos, content) ->{
            unitHelper.setNumeral(activity,pos);
        });
    }

    /**
     * 初始化Helper
     */
    private void initHelper() {
        unitHelper = new UnitHelper(activity);
        //开始请求
        unitHelper.setOnPreRequestListener(() -> {});
        //结束请求
        unitHelper.setOnFinishRequestListener(() -> {});
        //请求成功
        unitHelper.setOnSuccessRequestListener((object -> {}));
        //请求失败
        unitHelper.setOnFailedRequestListener((object -> toast(object.toString(),2000)));
        //温度单位列表回调
        unitHelper.setOnGetTempListener((select,arrayList) -> {
            tempAdapter.setItems(arrayList);
            tempAdapter.setSelect(select);
            tempAdapter.notifyDataSetChanged();
        });
        //压力单位列表回调
        unitHelper.setOnGetPressureListener((select,arrayList) -> {
            tireAdapter.setItems(arrayList);
            tireAdapter.setSelect(select);
            tireAdapter.notifyDataSetChanged();
        });
        //数字单位列表回调
        unitHelper.setOnGetNumeralListener((select,arrayList) -> {
            numeralAdapter.setItems(arrayList);
            numeralAdapter.setSelect(select);
            numeralAdapter.notifyDataSetChanged();
        });
        //读取单位列表
        unitHelper.getUnit(activity);
    }
}

File: app/src/main/java/com/orange/tpms/ue/kt_frag/Frag_Setting_Unit.kt
package com.orange.tpms.ue.kt_frag


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.orange.jzchi.jzframework.JzActivity
import com.orange.tpms.R
import com.orange.tpms.RootFragement
import com.orange.tpms.adapter.SelectAdapter
import com.orange.tpms.helper.UnitHelper

/**
 * A simple [Fragment] subclass.
 *
 */
class Frag_Setting_Unit : RootFragement(R.layout.fragment_frag__setting__unit) {
    override fun viewInit() {
        rvTemperature=rootview.findViewById(R.id.rv_temp)
        rvTirePressure=rootview.findViewById(R.id.rv_tire)
        rvNumeral=rootview.findViewById(R.id.rv_numeral)
        initView()
        initHelper()
    }

    lateinit var rvTemperature: androidx.recyclerview.widget.RecyclerView//列表
    lateinit var rvTirePressure: androidx.recyclerview.widget.RecyclerView//列表
    lateinit var rvNumeral: androidx.recyclerview.widget.RecyclerView//列表 lateinit var SelectAdapter tempAdapter;//Adapter
    lateinit var tempAdapter: SelectAdapter//Adapter
    lateinit var tireAdapter: SelectAdapter//Adapter
    lateinit var numeralAdapter: SelectAdapter//Adapter
    lateinit var tempLinearLayoutManager: androidx.recyclerview.widget.LinearLayoutManager//列表表格布局
    lateinit var tireLinearLayoutManager: androidx.recyclerview.widget.LinearLayoutManager//列表表格布局
    lateinit var numeralLinearLayoutManager: androidx.recyclerview.widget.LinearLayoutManager//列表表格布局
    lateinit var unitHelper: UnitHelper//Helper


    /**
     * 初始化页面
     */
    private fun initView() {
            tempLinearLayoutManager = androidx.recyclerview.widget.LinearLayoutManager(activity)
        rvTemperature.layoutManager = tempLinearLayoutManager
        tempAdapter = SelectAdapter(activity)
        rvTemperature.adapter = tempAdapter
        tempAdapter.setOnItemClickListener { pos, content -> unitHelper.setTemp(activity, pos)
            SetTem(pos)}
        //配置TireRecyclerView
            tireLinearLayoutManager = androidx.recyclerview.widget.LinearLayoutManager(activity)
        rvTirePressure.layoutManager = tireLinearLayoutManager
        tireAdapter = SelectAdapter(activity)
        rvTirePressure.adapter = tireAdapter
        tireAdapter.setOnItemClickListener { pos, content -> unitHelper.setPressure(activity, pos)
            SetPr(pos)}
        //配置NumeralRecyclerView
            numeralLinearLayoutManager = androidx.recyclerview.widget.LinearLayoutManager(activity)
        rvNumeral.layoutManager = numeralLinearLayoutManager
        numeralAdapter = SelectAdapter(activity)
        rvNumeral.adapter = numeralAdapter
        numeralAdapter.setOnItemClickListener { pos, content -> unitHelper.setNumeral(activity, pos)
            SetNs(pos)}
    }

    /**
     * 初始化Helper
     */
    private fun initHelper() {
        unitHelper = UnitHelper(activity)
        //开始请求
        unitHelper.setOnPreRequestListener { }
        //结束请求
        unitHelper.setOnFinishRequestListener { }
        //请求成功
        unitHelper.setOnSuccessRequestListener { `object` -> }
        //请求失败
        unitHelper.setOnFailedRequestListener { `object` -> JzActivity.getControlInstance().toast(`object`.toString()) }
        //温度单位列表回调
        unitHelper.setOnGetTempListener { select, arrayList ->
            tempAdapter.items = arrayList
            tempAdapter.setSelect(select)
            tempAdapter.notifyDataSetChanged()
        }
        //压力单位列表回调
        unitHelper.setOnGetPressureListener { select, arrayList ->
            tireAdapter.items = arrayList
            tireAdapter.setSelect(select)
            tireAdapter.notifyDataSetChanged()
        }
        //数字单位列表回调
        unitHelper.setOnGetNumeralListener { select, arrayList ->
            numeralAdapter.items = arrayList
            numeralAdapter.setSelect(select)
            numeralAdapter.notifyDataSetChanged()
        }
        //读取单位列表
        unitHelper.getUnit(activity)
    }
}
